src/main/java/org/opensearch/geospatial/geojson/FeatureCollection.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.geojson;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.opensearch.geospatial.GeospatialParser;
+
+/**
+ * FeatureCollection represents GEOJSON of type FeatureCollection. A FeatureCollection object has a member
+ * with the name "features".  The value of "features" is a List.
+ * Each element of the list is a {@link Feature} object.  It
+ * is possible for this list to be empty.
+ */
+public final class FeatureCollection {
+    public static final String TYPE = "FeatureCollection";
+    public static final String FEATURES_KEY = "features";
+    public static final String TYPE_KEY = "type";
+    private final List<Feature> features;
+
+    private FeatureCollection() {
+        this.features = new ArrayList<>();
+    }
+
+    /**
+     * Gets the list of Features
+     *
+     * @return List of {@link Feature}
+     */
+    public List<Feature> getFeatures() {
+        return Collections.unmodifiableList(features);
+    }
+
+    /**
+     * Add Features to this collection
+     * @param features List of {@link Feature}
+     */
+    public void addFeatures(List<Feature> features) {
+        Objects.requireNonNull(features);
+        this.features.addAll(features);
+    }
+
+    /**
+     * The static method to create an instance of {@link FeatureCollection} from input.
+     *
+     * @param input the object from where {@link FeatureCollection} will be extracted
+     * @return FeatureCollection Instance from given input
+     */
+    public static FeatureCollection create(final Map<String, Object> input) {
+        Object geoJSONType = input.get(TYPE_KEY);
```

Comment:
Do we need to add check for null for input variable?

Reply:
Added.